dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/RuleActionAssignValueImplementer.java

```diff
@@ -49,7 +49,7 @@
 {
     private static final Log log = LogFactory.getLog( RuleActionAssignValueImplementer.class );
 
-    private static final String REGEX = "\\w+";
+    private static final String REGEX = "[a-zA-Z0-9]+(?:[\\w -]*[a-zA-Z0-9]+)*";
```

Comment:
Great - web is a bit less restrictive, and also allows underscore and dot. Perhaps we can use the same regex as a basis?
/[A#CV]\{[\w -_.]+}/g